packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -837,7 +832,9 @@ abstract class CustomArrayNodeBase<const T extends ImplicitAllowedTypes>
 	): void {
 		const destinationField = getSequenceField(this);
 		if (destinationField.context === undefined) {
-			throw new UsageError(`An array cannot be mutated before being inserted into the tree`);
+			throw new UsageError(
+				`Array elements cannot be moved before being inserted into the tree`,
```

Comment:
This message makes it sound like the problem is that the elements being moved have to already be inserted, but judging by the code, it looks more like the destination node is the problem. I'm guessing they're both potentially a problem.